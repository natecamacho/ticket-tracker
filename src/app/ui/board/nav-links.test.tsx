import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import NavLinks from "./nav-links";

describe("NavLinks component", () => {
  it("should render NavLinks unchanged", () => {
    const { container } = render(<NavLinks />);

    expect(screen.getByText("Board")).toBeInTheDocument();
    expect(screen.getByText("Tickets")).toBeInTheDocument();
    expect(screen.getByText("Backlog")).toBeInTheDocument();

    expect(screen.getAllByRole("link").length).toEqual(3);
    expect(screen.getByRole("link", { name: /Board/i })).toHaveAttribute(
      "href",
      "/board"
    );
    expect(screen.getByRole("link", { name: /Tickets/i })).toHaveAttribute(
      "href",
      "/tickets"
    );
    expect(screen.getByRole("link", { name: /Backlog/i })).toHaveAttribute(
      "href",
      "/backlog"
    );

    expect(container).toMatchSnapshot();
  });
});
