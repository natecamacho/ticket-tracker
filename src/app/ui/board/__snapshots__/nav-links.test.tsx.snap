// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks component should render NavLinks unchanged 1`] = `
<div>
  <a
    class="flex h-[48px] flex-none justify-center p-2 px-3 items-center gap-2 rounded-md bg-gray-50 text-sm font-medium hover:bg-sky-100 hover:text-blue-600"
    href="/board"
  >
    <p
      class="block"
    >
      Board
    </p>
  </a>
  <a
    class="flex h-[48px] flex-none justify-center p-2 px-3 items-center gap-2 rounded-md bg-gray-50 text-sm font-medium hover:bg-sky-100 hover:text-blue-600"
    href="/tickets"
  >
    <p
      class="block"
    >
      Tickets
    </p>
  </a>
  <a
    class="flex h-[48px] flex-none justify-center p-2 px-3 items-center gap-2 rounded-md bg-gray-50 text-sm font-medium hover:bg-sky-100 hover:text-blue-600"
    href="/backlog"
  >
    <p
      class="block"
    >
      Backlog
    </p>
  </a>
</div>
`;
